pkg/controlplane/controller.go

```diff
@@ -163,9 +163,7 @@ func (c *Controller) Start() {
 
 	// Reconcile during first run removing itself until server is ready.
 	endpointPorts := createEndpointPortSpec(c.PublicServicePort, "https", c.ExtraEndpointPorts)
-	if err := c.EndpointReconciler.RemoveEndpoints(kubernetesServiceName, c.PublicIP, endpointPorts); err == nil {
-		klog.Error("Found stale data, removed previous endpoints on kubernetes service, apiserver didn't exit successfully previously")
-	} else if !storage.IsNotFound(err) {
+	if err := c.EndpointReconciler.RemoveEndpoints(kubernetesServiceName, c.PublicIP, endpointPorts); err != nil && !storage.IsNotFound(err) {
```

Comment:
@aojea wasn't that your code?

Reply:
See the commit message: https://github.com/kubernetes/kubernetes/pull/107677/commits/be4bf5cb20c18fa9b74e5aff7fc64639b1797bd3

Reply:
@aojea : does that explain it?

Reply:
heh, I understand now, this thing bit me twice this week
The test is using the masterCount reconciler, not the lease reconciler. The lease reconciler is the default ... 
The integration framework should use the lease reconciler ... I think I will write something to try to deprecate the masterCount reconciler, it is flaky :(

Reply:
https://github.com/kubernetes/kubernetes/blob/76cdb57ccfbfebc689fbce45f289add8a0562e07/cmd/kube-apiserver/app/options/options.go#L114

Reply:
I think we should keep the code and deprecate the master-count
https://github.com/kubernetes/kubernetes/pull/107952

Reply:
As @aojea is going to address the false error message, I took this commit out of this PR.

Reply:
The PR was closed without a conclusion, but the issue about the error message is still open: https://github.com/kubernetes/kubernetes/issues/107953